Replace more use of MallocPtr with MallocSpan

cdumez opened the pull request:
#### 419636cf5d236be22ae30d2e4a9b62360b9a2848
<pre>
Replace more use of MallocPtr with MallocSpan
<a href="https://bugs.webkit.org/show_bug.cgi?id=282521">https://bugs.webkit.org/show_bug.cgi?id=282521</a>

Reviewed by Darin Adler.

* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
(JSC::ExecutableMemoryHandle::~ExecutableMemoryHandle):
* Source/JavaScriptCore/runtime/CachePayload.cpp:
(JSC::CachePayload::makeMallocPayload):
(JSC::CachePayload::makeEmptyPayload):
(JSC::CachePayload::CachePayload):
(JSC::CachePayload::span const):
* Source/JavaScriptCore/runtime/CachePayload.h:
* Source/JavaScriptCore/runtime/CachedBytecode.h:
(JSC::CachedBytecode::create):
* Source/JavaScriptCore/runtime/CachedTypes.cpp:
(JSC::Encoder::release):
(JSC::Encoder::Page::Page):
(JSC::Encoder::Page::malloc):
(JSC::Encoder::Page::buffer const):
(JSC::Encoder::Page::buffer):
(JSC::Encoder::Page::mutableSpan):
(JSC::Encoder::Page::span const):
(JSC::Encoder::Page::getOffset const):
(JSC::Encoder::Page::alignEnd):
(JSC::Encoder::Page::capacity const):
* Source/JavaScriptCore/wasm/js/JSWebAssemblyGlobal.h:
* Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.cpp:
* Source/JavaScriptCore/wasm/js/JSWebAssemblyTable.h:
* Source/WTF/wtf/win/MemoryFootprintWin.cpp:
(WTF::memoryFootprint):
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedTileBuffer.cpp:
(WebCore::CoordinatedUnacceleratedTileBuffer::CoordinatedUnacceleratedTileBuffer):
(WebCore::CoordinatedUnacceleratedTileBuffer::tryEnsureSurface):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedTileBuffer.h:
* Source/WebGPU/WGSL/AST/ASTBuilder.h:
* Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp:
(IPC::readBytesFromSocket):
(IPC::Connection::sendOutputMessage):
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
(WebKit::encodeLegacySessionState):

Canonical link: <a href="https://commits.webkit.org/286085@main">https://commits.webkit.org/286085@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/15c2b5f19628fe7996aae7690445f5498a71fee2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74761 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54191 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27578 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/79184 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25999 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/76878 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63327 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1976 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58722 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17005 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77828 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48872 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64242 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39116 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/46119 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21739 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-masking/clip-path/animations/clip-path-animation-font-size-mixed-change.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24332 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/67898 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67311 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22084 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80674 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/74019 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/2079 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1234 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66985 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/2227 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64260 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66276 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10216 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8366 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/96290 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11539 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2044 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/4831 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21054 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/2072 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/2992 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/2079 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/runtime/CachePayload.cpp:
```diff
@@ -35,23 +35,23 @@ CachePayload CachePayload::makeMappedPayload(FileSystem::MappedFileData&& data)
 CachePayload CachePayload::makeEmptyPayload()
 {
-    return CachePayload(std::pair { nullptr, 0 });
+    return CachePayload(MallocSpan<uint8_t, VMMalloc>());
```
Can't just use `{ }`?

darinadler replied:
Can't just use `{ }`?

darinadler commented on Source/JavaScriptCore/runtime/CachePayload.cpp:
```diff
@@ -35,23 +35,23 @@ CachePayload CachePayload::makeMappedPayload(FileSystem::MappedFileData&& data)
 {
     m_data = WTFMove(other.m_data);
-    other.m_data = std::pair { nullptr, 0 };
+    other.m_data = MallocSpan<uint8_t, VMMalloc>();
```
Should use `std::exchange`.

cdumez committed:
Replace more use of MallocPtr with MallocSpan
https://bugs.webkit.org/show_bug.cgi?id=282521

Reviewed by Darin Adler.

* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
(JSC::ExecutableMemoryHandle::~ExecutableMemoryHandle):
* Source/JavaScriptCore/runtime/CachePayload.cpp:
(JSC::CachePayload::makeMallocPayload):
(JSC::CachePayload::makeEmptyPayload):
(JSC::CachePayload::CachePayload):
(JSC::CachePayload::span const):
* Source/JavaScriptCore/runtime/CachePayload.h:
* Source/JavaScriptCore/runtime/CachedBytecode.h:
(JSC::CachedBytecode::create):
* Source/JavaScriptCore/runtime/CachedTypes.cpp:
(JSC::Encoder::release):
(JSC::Encoder::Page::Page):
(JSC::Encoder::Page::malloc):
(JSC::Encoder::Page::buffer const):
(JSC::Encoder::Page::buffer):
(JSC::Encoder::Page::mutableSpan):
(JSC::Encoder::Page::span const):
(JSC::Encoder::Page::getOffset const):
(JSC::Encoder::Page::alignEnd):
(JSC::Encoder::Page::capacity const):
* Source/JavaScriptCore/wasm/js/JSWebAssemblyGlobal.h:
* Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.cpp:
* Source/JavaScriptCore/wasm/js/JSWebAssemblyTable.h:
* Source/WTF/wtf/win/MemoryFootprintWin.cpp:
(WTF::memoryFootprint):
* Source/WebCore/platform/Length.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedTileBuffer.cpp:
(WebCore::CoordinatedUnacceleratedTileBuffer::CoordinatedUnacceleratedTileBuffer):
(WebCore::CoordinatedUnacceleratedTileBuffer::tryEnsureSurface):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedTileBuffer.h:
* Source/WebGPU/WGSL/AST/ASTBuilder.h:
* Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp:
(IPC::readBytesFromSocket):
(IPC::Connection::sendOutputMessage):
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
(WebKit::encodeLegacySessionState):

Canonical link: https://commits.webkit.org/286085@main
